staging/src/k8s.io/client-go/transport/transport.go

```diff
@@ -167,3 +169,15 @@ func rootCertPool(caData []byte) *x509.CertPool {
 	certPool.AppendCertsFromPEM(caData)
 	return certPool
 }
+
+func tryCancelRequest(rt http.RoundTripper, req *http.Request) {
+	type canceler interface {
+		CancelRequest(*http.Request)
+	}
+	switch rt := rt.(type) {
+	case canceler:
+		rt.CancelRequest(req)
+	case utilnet.RoundTripperWrapper:
```

Comment:
do you need a 'default' case, that is neither canceler or utilnet.RoundTripperWrapper?

Reply:
What do you think we should do other than give up silently? Log an error?

Reply:
I think log an error is good. since it is something unexpected.